diff -Naur icu-native/source/common/putilimp.h icu/source/common/putilimp.h
--- icu-native/source/common/putilimp.h	2017-02-03 23:12:43.468839423 +0100
+++ icu/source/common/putilimp.h	2017-02-03 23:12:24.501980770 +0100
@@ -103,6 +103,7 @@
 #   define U_NL_LANGINFO_CODESET CODESET
 #endif
 
+#if 0
 #ifdef U_TZSET
     /* Use the predefined value. */
 #elif U_PLATFORM_USES_ONLY_WIN32_API
@@ -147,6 +148,7 @@
 #else
 #   define U_TZNAME tzname
 #endif
+#endif
 
 #ifdef U_HAVE_MMAP
     /* Use the predefined value. */
diff -Naur icu-native/source/i18n/digitlst.cpp icu/source/i18n/digitlst.cpp
--- icu-native/source/i18n/digitlst.cpp	2017-02-03 23:12:43.458839321 +0100
+++ icu/source/i18n/digitlst.cpp	2017-02-03 23:20:49.026494913 +0100
@@ -50,6 +50,7 @@
 #include <stdio.h>
 #include <limits>
 
+#if 0
 #if !defined(U_USE_STRTOD_L)
 # if U_PLATFORM_USES_ONLY_WIN32_API
 #   define U_USE_STRTOD_L 1
@@ -59,6 +60,9 @@
 #   define U_USE_STRTOD_L 0
 # endif
 #endif
+#endif
+
+#define U_USE_STRTOD_L 0
 
 #if U_USE_STRTOD_L && !U_PLATFORM_USES_ONLY_WIN32_API
 # if U_PLATFORM == U_PF_CYGWIN

