--- vitasdk/share/vita.toolchain.cmake.orig	2025-07-20 15:04:17.000000000 +0200
+++ vitasdk/share/vita.toolchain.cmake	2025-10-21 20:34:29.213607535 +0200
@@ -109,6 +109,10 @@
 set( CMAKE_SYSTEM_PREFIX_PATH ${CMAKE_FIND_ROOT_PATH})
 set( CMAKE_INSTALL_PREFIX "${VITASDK}/arm-vita-eabi" CACHE PATH "default install path" )
 set( PKG_CONFIG_EXECUTABLE "${VITASDK}/bin/arm-vita-eabi-pkg-config" CACHE PATH "Path of pkg-config executable" )
+if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.16)
+  set(CMAKE_FIND_USE_SYSTEM_ENVIRONMENT_PATH OFF)
+  string(REPLACE ":" ";" CMAKE_SYSTEM_PROGRAM_PATH "$ENV{PATH}")
+endif()
 
 # only search for libraries and includes in vita toolchain
 if( NOT CMAKE_FIND_ROOT_PATH_MODE_LIBRARY )
