diff --color -Naur wildmidi-wildmidi-0.4.1.orig/src/file_io.c wildmidi-wildmidi-0.4.1/src/file_io.c
--- wildmidi-wildmidi-0.4.1.orig/src/file_io.c	2017-02-26 18:30:45.000000000 +0100
+++ wildmidi-wildmidi-0.4.1/src/file_io.c	2018-02-11 22:51:24.484527487 +0100
@@ -127,7 +127,7 @@
 #elif defined(WILDMIDI_AMIGA)
     BPTR buffer_fd;
     long filsize;
-#elif defined(_3DS) || defined(GEKKO)
+#elif defined(_3DS) || defined(GEKKO) || defined(__vita__)
     int buffer_fd;
     struct stat buffer_stat;
 #else /* unix builds */
diff --color -Naur wildmidi-wildmidi-0.4.1.orig/src/lock.c wildmidi-wildmidi-0.4.1/src/lock.c
--- wildmidi-wildmidi-0.4.1.orig/src/lock.c	2017-02-26 18:30:45.000000000 +0100
+++ wildmidi-wildmidi-0.4.1/src/lock.c	2018-02-11 23:15:03.135066944 +0100
@@ -33,6 +33,8 @@
 #include <os2.h>
 #elif defined(WILDMIDI_AMIGA)
 #include <proto/dos.h>
+#elif defined(__vita__)
+#include <psp2/kernel/processmgr.h>
 #else /* unixish ... */
 #define _GNU_SOURCE
 #include <unistd.h> /* usleep() */
@@ -70,6 +72,8 @@
     DosSleep(10);
 #elif defined(WILDMIDI_AMIGA)
     Delay(1);
+#elif defined(__vita__)
+    sceKernelDelayThread(500);
 #else
     usleep(500);
 #endif
