diff -Naur libsndfile-1.0.27-orig/configure.ac libsndfile-1.0.27/configure.ac
--- libsndfile-1.0.27-orig/configure.ac	2016-06-21 19:15:29.548606789 +0200
+++ libsndfile-1.0.27/configure.ac	2016-06-21 19:32:54.258549707 +0200
@@ -227,6 +227,10 @@
 		;;
 	esac

+TYPEOF_SF_COUNT_T="int64_t"
+SIZEOF_SF_COUNT_T=8
+SF_COUNT_MAX="0x7FFFFFFFFFFFFFFFLL"
+
 if test $SIZEOF_SF_COUNT_T = 4 ; then
 	SF_COUNT_MAX="0x7FFFFFFF"
 	fi
diff -Naur libsndfile-1.0.27-orig/Makefile.am libsndfile-1.0.27/Makefile.am
--- libsndfile-1.0.27-orig/src/GSM610/gsm610_priv.h	2016-06-21 19:15:29.545273456 +0200
+++ libsndfile-1.0.27/src/GSM610/gsm610_priv.h	2016-06-21 19:32:50.715216567 +0200
@@ -18,12 +18,12 @@
 /* Added by Erik de Castro Lopo */


+#include <stdint.h>
+//typedef short				int16_t ;		/* 16 bit signed int	*/
+//typedef int					int32_t ;	/* 32 bit signed int	*/

-typedef short				int16_t ;		/* 16 bit signed int	*/
-typedef int					int32_t ;	/* 32 bit signed int	*/
-
-typedef unsigned short		uint16_t ;		/* unsigned int16_t	*/
-typedef unsigned int		uint32_t ;	/* unsigned int32_t	*/
+//typedef unsigned short		uint16_t ;		/* unsigned int16_t	*/
+//typedef unsigned int		uint32_t ;	/* unsigned int32_t	*/

 struct gsm_state
 {	int16_t			dp0 [280] ;
diff -Naur libsndfile-1.0.27-orig/src/wavlike.c libsndfile-1.0.27/src/wavlike.c
--- libsndfile-1.0.27-orig/src/wavlike.c	2016-06-21 19:15:29.545273456 +0200
+++ libsndfile-1.0.27/src/wavlike.c	2016-06-21 19:34:12.721878753 +0200
@@ -700,7 +700,7 @@
 } /* srate2blocksize */

 int
-wavlike_read_bext_chunk (SF_PRIVATE *psf, uint32_t chunksize)
+wavlike_read_bext_chunk (SF_PRIVATE *psf, unsigned int chunksize)
 {
 	SF_BROADCAST_INFO_16K * b ;
 	uint32_t bytes = 0 ;
@@ -787,7 +787,7 @@
 } /* wavlike_write_bext_chunk */

 int
-wavlike_read_cart_chunk (SF_PRIVATE *psf, uint32_t chunksize)
+wavlike_read_cart_chunk (SF_PRIVATE *psf, unsigned int chunksize)
 {	SF_CART_INFO_16K *c ;
 	uint32_t bytes = 0 ;
 	int k ;
diff -Naur libsndfile-1.0.28-orig/src/common.c libsndfile-1.0.28/src/common.c
--- libsndfile-1.0.28-orig/src/common.c	2016-06-21 17:45:26.715568661 +0200
+++ libsndfile-1.0.28/src/common.c	2016-06-21 17:46:34.448898294 +0200
@@ -1674,7 +1674,7 @@

 FILE *
 psf_open_tmpfile (char * fname, size_t fnamelen)
-{	const char * tmpdir ;
+{	/*const char * tmpdir ;
 	FILE * file ;

 	if (OS_IS_WIN32)
@@ -1694,6 +1694,6 @@
 	if ((file = fopen (fname, "wb+")) != NULL)
 		return file ;

-	memset (fname, 0, fnamelen) ;
+	memset (fname, 0, fnamelen) ;*/
 	return NULL ;
 } /* psf_open_tmpfile */
