diff -Naur libsndfile-1.0.28-orig/src/common.c libsndfile-1.0.28/src/common.c
--- libsndfile-1.0.28-orig/src/common.c	2016-06-21 17:45:26.715568661 +0200
+++ libsndfile-1.0.28/src/common.c	2016-06-21 17:46:34.448898294 +0200
@@ -1674,7 +1674,7 @@

 FILE *
 psf_open_tmpfile (char * fname, size_t fnamelen)
-{	const char * tmpdir ;
+{	/*const char * tmpdir ;
 	FILE * file ;

 	if (OS_IS_WIN32)
@@ -1694,6 +1694,6 @@
 	if ((file = fopen (fname, "wb+")) != NULL)
 		return file ;

-	memset (fname, 0, fnamelen) ;
+	memset (fname, 0, fnamelen) ;*/
 	return NULL ;
 } /* psf_open_tmpfile */
