diff -Naur pixman-0.36.0-orig/pixman/pixman-arm.c pixman-0.36.0/pixman/pixman-arm.c
--- pixman-0.36.0-orig/pixman/pixman-arm.c	2020-04-02 16:49:09.909533624 +0200
+++ pixman-0.36.0/pixman/pixman-arm.c	2020-04-02 16:50:44.279533318 +0200
@@ -176,6 +176,30 @@
     return features;
 }
 
+#elif defined (_3DS) /* 3DS homebrew (devkitARM) */
+
+static arm_cpu_features_t
+detect_cpu_features (void)
+{
+	arm_cpu_features_t features = 0;
+
+	features |= ARM_V6;
+
+	return features;
+}
+
+#elif defined (PSP2) || defined (__SWITCH__) /* Vita (VitaSDK) or Switch (devkitA64) homebrew */
+
+static arm_cpu_features_t
+detect_cpu_features (void)
+{
+	arm_cpu_features_t features = 0;
+
+	features |= ARM_NEON;
+
+	return features;
+}
+
 #else /* Unknown */
 
 static arm_cpu_features_t
