diff -Naur lpp-3ds_libraries-orig/Makefile lpp-3ds_libraries/Makefile
--- lpp-3ds_libraries-orig/Makefile	2016-04-21 18:13:19.160028000 +0000
+++ lpp-3ds_libraries/Makefile	2016-04-21 18:13:53.366028000 +0000
@@ -16,18 +16,18 @@
 # DATA is a list of directories containing data files
 # INCLUDES is a list of directories containing header files
 #---------------------------------------------------------------------------------
-TARGET		:=	generic_lib
+TARGET		:=	khax
 BUILD		:=	build
-SOURCES		:=	source
-DATA		:=	data
+SOURCES		:=	source_libkhax
+DATA		:=	nodata
 INCLUDES	:=	include
 
 #---------------------------------------------------------------------------------
 # options for code generation
 #---------------------------------------------------------------------------------
-ARCH	:=	-march=armv6k -mtune=mpcore -mfloat-abi=softfp
+ARCH	:=	-march=armv6k -mtune=mpcore -mfloat-abi=hard
 
-CFLAGS	:=	-g -O2 -mword-relocations \
+CFLAGS	:=	-g0 -O2 -mword-relocations \
 			-fomit-frame-pointer -ffast-math \
 			$(ARCH)
 
