diff -Naur libxmp-lite-4.4.0.orig/config.sub libxmp-lite-4.4.0/config.sub
--- libxmp-lite-4.4.0.orig/config.sub	2016-02-09 12:35:06.000000000 +0100
+++ libxmp-lite-4.4.0/config.sub	2016-09-04 11:15:51.481189993 +0200
@@ -123,13 +123,17 @@
 # Here we must recognize all the valid KERNEL-OS combinations.
 maybe_os=`echo $1 | sed 's/^\(.*\)-\([^-]*-[^-]*\)$/\2/'`
 case $maybe_os in
-  nto-qnx* | linux-gnu* | linux-dietlibc | linux-newlib* | linux-uclibc* | \
+  nto-qnx* | linux-gnu* | linux-android* | linux-dietlibc | linux-newlib* | linux-uclibc* | \
   uclinux-uclibc* | uclinux-gnu* | kfreebsd*-gnu* | knetbsd*-gnu* | netbsd*-gnu* | \
   kopensolaris*-gnu* | \
   storm-chaos* | os2-emx* | rtmk-nova*)
     os=-$maybe_os
     basic_machine=`echo $1 | sed 's/^\(.*\)-\([^-]*-[^-]*\)$/\1/'`
     ;;
+  android-linux)
+    os=-linux-android
+    basic_machine=`echo $1 | sed 's/^\(.*\)-\([^-]*-[^-]*\)$/\1/'`-unknown
+    ;;
   *)
     basic_machine=`echo $1 | sed 's/-[^-]*$//'`
     if [ $basic_machine != $1 ]
@@ -1294,7 +1298,7 @@
 	      | -udi* | -eabi* | -lites* | -ieee* | -go32* | -aux* \
 	      | -chorusos* | -chorusrdb* | -cegcc* \
 	      | -cygwin* | -pe* | -psos* | -moss* | -proelf* | -rtems* \
-	      | -mingw32* | -linux-gnu* | -linux-newlib* | -linux-uclibc* \
+	      | -mingw32* | -linux-gnu* | -linux-newlib* | -linux-uclibc* | -linux-android* \
 	      | -uxpv* | -beos* | -mpeix* | -udk* \
 	      | -interix* | -uwin* | -mks* | -rhapsody* | -darwin* | -opened* \
 	      | -openstep* | -oskit* | -conix* | -pw32* | -nonstopux* \
diff -Naur libxmp-lite-4.4.0.orig/src/common.h libxmp-lite-4.4.0/src/common.h
--- libxmp-lite-4.4.0.orig/src/common.h	2016-07-16 13:37:36.000000000 +0200
+++ libxmp-lite-4.4.0/src/common.h	2016-09-04 11:11:49.103931106 +0200
@@ -10,7 +10,7 @@
 #include "xmp.h"
 
 #if defined(__GNUC__) || defined(__clang__)
-#if !defined(WIN32) && !defined(ANDROID) && !defined(__APPLE__) && !defined(__AMIGA__) && !defined(B_BEOS_VERSION) && !defined(__ATHEOS__) && !defined(EMSCRIPTEN) && !defined(__MINT__) 
+#if !defined(WIN32) && !defined(__ANDROID__) && !defined(__APPLE__) && !defined(__AMIGA__) && !defined(B_BEOS_VERSION) && !defined(__ATHEOS__) && !defined(EMSCRIPTEN) && !defined(__MINT__) 
 #define USE_VERSIONED_SYMBOLS
 #endif
 #endif
@@ -92,7 +92,7 @@
 #endif
 #endif
 
-#elif defined ANDROID
+#elif defined __ANDROID__
 
 #ifdef DEBUG
 #include <android/log.h>
diff -Naur libxmp-lite-4.4.0.orig/src/load.c libxmp-lite-4.4.0/src/load.c
--- libxmp-lite-4.4.0.orig/src/load.c	2016-07-19 14:29:42.000000000 +0200
+++ libxmp-lite-4.4.0/src/load.c	2016-09-04 11:11:27.356912025 +0200
@@ -171,7 +171,7 @@
 		return 0;
 	}
 
-#if defined ANDROID || defined __native_client__
+#if defined __ANDROID__ || defined __native_client__
 	/* Don't use external helpers in android */
 	if (cmd) {
 		return 0;
diff -Naur libxmp-lite-4.4.0.orig/src/load_helpers.c libxmp-lite-4.4.0/src/load_helpers.c
--- libxmp-lite-4.4.0.orig/src/load_helpers.c	2016-07-19 14:33:00.000000000 +0200
+++ libxmp-lite-4.4.0/src/load_helpers.c	2016-09-04 11:11:33.633680411 +0200
@@ -30,7 +30,7 @@
 
 #ifndef LIBXMP_CORE_PLAYER
 
-#ifdef ANDROID
+#ifdef __ANDROID__
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <unistd.h>
