diff -Naur .\freetype-2.6.5-old/src/gzip/infcodes.h .\freetype-2.6.5/src/gzip/infcodes.h
--- .\freetype-2.6.5-old/src/gzip/infcodes.h	2016-01-12 22:00:40 +0100
+++ .\freetype-2.6.5/src/gzip/infcodes.h	2017-04-20 02:39:57 +0200
@@ -14,17 +14,17 @@
 struct inflate_codes_state;
 typedef struct inflate_codes_state FAR inflate_codes_statef;
 
-local inflate_codes_statef *inflate_codes_new OF((
+inflate_codes_statef *inflate_codes_new OF((
     uInt, uInt,
     inflate_huft *, inflate_huft *,
     z_streamp ));
 
-local int inflate_codes OF((
+int inflate_codes OF((
     inflate_blocks_statef *,
     z_streamp ,
     int));
 
-local void inflate_codes_free OF((
+void inflate_codes_free OF((
     inflate_codes_statef *,
     z_streamp ));
 
diff -Naur .\freetype-2.6.5-old/src/gzip/inftrees.h .\freetype-2.6.5/src/gzip/inftrees.h
--- .\freetype-2.6.5-old/src/gzip/inftrees.h	2016-01-12 22:00:40 +0100
+++ .\freetype-2.6.5/src/gzip/inftrees.h	2017-04-20 02:39:04 +0200
@@ -35,14 +35,14 @@
    value below is more than safe. */
 #define MANY 1440
 
-local  int inflate_trees_bits OF((
+int inflate_trees_bits OF((
     uIntf *,                    /* 19 code lengths */
     uIntf *,                    /* bits tree desired/actual depth */
     inflate_huft * FAR *,       /* bits tree result */
     inflate_huft *,             /* space for trees */
     z_streamp));                /* for messages */
 
-local  int inflate_trees_dynamic OF((
+int inflate_trees_dynamic OF((
     uInt,                       /* number of literal/length codes */
     uInt,                       /* number of distance codes */
     uIntf *,                    /* that many (total) code lengths */
@@ -53,7 +53,7 @@
     inflate_huft *,             /* space for trees */
     z_streamp));                /* for messages */
 
-local  int inflate_trees_fixed OF((
+int inflate_trees_fixed OF((
     uIntf *,                    /* literal desired/actual bit depth */
     uIntf *,                    /* distance desired/actual bit depth */
     const inflate_huft * FAR *, /* literal/length tree result */
diff -Naur .\freetype-2.6.5-old/src/gzip/infutil.h .\freetype-2.6.5/src/gzip/infutil.h
--- .\freetype-2.6.5-old/src/gzip/infutil.h	2016-01-12 22:00:40 +0100
+++ .\freetype-2.6.5/src/gzip/infutil.h	2017-04-20 02:39:37 +0200
@@ -86,11 +86,11 @@
 
 /* masks for lower bits (size given to avoid silly warnings with Visual C++) */
 #ifndef NO_INFLATE_MASK
-local uInt inflate_mask[17];
+local const uInt inflate_mask[17];
 #endif
 
 /* copy as much as possible from the sliding window to the output area */
-local int inflate_flush OF((
+int inflate_flush OF((
     inflate_blocks_statef *,
     z_streamp ,
     int));
diff -Naur .\freetype-2.6.5-old/src/gzip/zutil.h .\freetype-2.6.5/src/gzip/zutil.h
--- .\freetype-2.6.5-old/src/gzip/zutil.h	2016-01-12 22:00:40 +0100
+++ .\freetype-2.6.5/src/gzip/zutil.h	2017-04-20 02:36:53 +0200
@@ -182,24 +182,13 @@
 #endif
 
 /* Diagnostic functions */
-#ifdef DEBUG
-#  include <stdio.h>
-   extern int z_verbose;
-   extern void z_error    OF((char *m));
-#  define Assert(cond,msg) {if(!(cond)) z_error(msg);}
-#  define Trace(x) {if (z_verbose>=0) fprintf x ;}
-#  define Tracev(x) {if (z_verbose>0) fprintf x ;}
-#  define Tracevv(x) {if (z_verbose>1) fprintf x ;}
-#  define Tracec(c,x) {if (z_verbose>0 && (c)) fprintf x ;}
-#  define Tracecv(c,x) {if (z_verbose>1 && (c)) fprintf x ;}
-#else
 #  define Assert(cond,msg)
 #  define Trace(x)
 #  define Tracev(x)
 #  define Tracevv(x)
 #  define Tracec(c,x)
 #  define Tracecv(c,x)
-#endif
+
 
 
 typedef uLong (*check_func) OF((uLong check, const Bytef *buf,
