diff -Naur .\icu-old/source/tools/escapesrc/escapesrc.cpp .\icu/source/tools/escapesrc/escapesrc.cpp 
--- .\icu-old/source/tools/escapesrc/escapesrc.cpp	2014-03-09 23:49:07 +0000
+++ .\icu/source/tools/escapesrc/escapesrc.cpp 	2017-04-19 23:08:14 +0000
@@ -4,7 +4,7 @@
 #include <stdio.h>
 #include <string>
 #include <stdlib.h>
-#include <unistd.h>
+//#include <unistd.h>
 #include <errno.h>
 #include <string.h>
 #include <iostream>
