diff -Naur .\libxmp-lite-4.4.1/src/common.h .\libxmp-lite-4.4.1-orig/src/common.h
--- .\libxmp-lite-4.4.1/src/common.h	2016-07-16 13:37:36 +0200
+++ .\libxmp-lite-4.4.1-orig/src/common.h	2017-05-23 00:52:17 +0200
@@ -78,19 +78,12 @@
 #ifndef CLIB_DECL
 #define CLIB_DECL
 #endif
-#ifdef DEBUG
-#ifndef ATTR_PRINTF
-#define ATTR_PRINTF(x,y)
-#endif
-void CLIB_DECL D_(const char *text, ...) ATTR_PRINTF(1,2);
-#else
 // VS prior to VC7.1 does not support variadic macros. VC8.0 does not optimize unused parameters passing
 #if _MSC_VER < 1400
 void __inline CLIB_DECL D_(const char *text, ...) { do {} while (0); }
 #else
 #define D_(args, ...) do {} while (0)
 #endif
-#endif
 
 #elif defined ANDROID

diff --git a/lite/src/format.c b/lite/src/format.c
--- a/libxmp-lite-4.4.1/src/format.c
+++ b/libxmp-lite-4.4.1-orig/src/format.c
@@ -27,20 +27,20 @@
 #endif
 #include "format.h"
 
-extern const struct format_loader xm_loader;
-extern const struct format_loader mod_loader;
-extern const struct format_loader it_loader;
-extern const struct format_loader s3m_loader;
+extern const struct format_loader libxmp_loader_xm;
+extern const struct format_loader libxmp_loader_mod;
+extern const struct format_loader libxmp_loader_it;
+extern const struct format_loader libxmp_loader_s3m;
 
 extern const struct pw_format *const pw_format[];
 
 const struct format_loader *const format_loader[5] = {
-	&xm_loader,
-	&mod_loader,
+	&libxmp_loader_xm,
+	&libxmp_loader_mod,
 #ifndef LIBXMP_CORE_DISABLE_IT
-	&it_loader,
+	&libxmp_loader_it,
 #endif
-	&s3m_loader,
+	&libxmp_loader_s3m,
 	NULL
 };
 
diff --git a/lite/src/loaders/mod_load.c b/lite/src/loaders/mod_load.c
--- a/libxmp-lite-4.4.1/src/loaders/mod_load.c
+++ b/libxmp-lite-4.4.1-orig/src/loaders/mod_load.c
@@ -36,7 +36,7 @@
 static int mod_test (HIO_HANDLE *, char *, const int);
 static int mod_load (struct module_data *, HIO_HANDLE *, const int);
 
-const struct format_loader mod_loader = {
+const struct format_loader libxmp_loader_mod = {
     "Protracker",
     mod_test,
     mod_load
