diff -Naur .\SDL2-2.0.5/src/core/windows/SDL_windows.c .\SDL2-2.0.5-orig/src/core/windows/SDL_windows.c
--- .\SDL2-2.0.5/src/core/windows/SDL_windows.c	2016-10-20 05:56:26 +0200
+++ .\SDL2-2.0.5-orig/src/core/windows/SDL_windows.c	2017-05-20 22:39:32 +0200
@@ -146,7 +146,7 @@
 char *
 WIN_LookupAudioDeviceName(const WCHAR *name, const GUID *guid)
 {
-#if __WINRT__
+#if defined(__WINRT__) || defined(_ARM_)
     return WIN_StringToUTF8(name);  /* No registry access on WinRT/UWP, go with what we've got. */
 #else
     static const GUID nullguid = { 0 };
diff -Naur .\SDL2-2.0.5/src/dynapi/SDL_dynapi.h .\SDL2-2.0.5-orig/src/dynapi/SDL_dynapi.h
--- .\SDL2-2.0.5/src/dynapi/SDL_dynapi.h	2016-10-20 05:56:26 +0200
+++ .\SDL2-2.0.5-orig/src/dynapi/SDL_dynapi.h	2017-05-20 19:57:18 +0200
@@ -59,7 +59,7 @@
 
 /* everyone else. This is where we turn on the API if nothing forced it off. */
 #ifndef SDL_DYNAMIC_API
-#define SDL_DYNAMIC_API 1
+#define SDL_DYNAMIC_API 0
 #endif
 
 #endif
