diff -Naur libmodplug-0.8.8.5-orig/src/load_abc.cpp libmodplug-0.8.8.5/src/load_abc.cpp
--- libmodplug-0.8.8.5-orig/src/load_abc.cpp	2016-04-18 09:06:59.678165911 +0000
+++ libmodplug-0.8.8.5/src/load_abc.cpp	2016-04-18 09:07:28.704257619 +0000
@@ -2418,8 +2418,8 @@
 			}
 		}
 		else {
-			srandom((uint32_t)time(0));	// initialize random generator with seed
-			retval->pickrandom = 1+(int)(10000.0*random()/(RAND_MAX+1.0));
+			srand((uint32_t)time(0));	// initialize random generator with seed
+			retval->pickrandom = 1+(int)(10000.0*rand()/(RAND_MAX+1.0));
 			// can handle pickin' from songbooks with 10.000 songs
 #ifdef NEWMIKMOD
 			sprintf(buf,"-%ld",retval->pickrandom+1); // next in sequence
