diff -Naur sdl-wii-2018-08-31/SDL/include/SDL_config_wii.h sdl-wii/SDL/include/SDL_config_wii.h
--- sdl-wii-2018-08-31/SDL/include/SDL_config_wii.h	2018-08-31 18:49:59.000000000 +0200
+++ sdl-wii/SDL/include/SDL_config_wii.h	2019-01-28 20:20:10.559626681 +0100
@@ -27,18 +27,6 @@
 
 #include <stdarg.h>
 
-/* Types */
-typedef signed char int8_t;
-typedef unsigned char uint8_t;
-typedef signed short int16_t;
-typedef unsigned short uint16_t;
-typedef signed long int32_t;
-typedef unsigned long uint32_t;
-typedef signed long long int64_t;
-typedef unsigned long long uint64_t;
-typedef unsigned int size_t;
-typedef unsigned int uintptr_t;
-
 /* Architecture */
 #define SDL_BYTEORDER		SDL_BIG_ENDIAN
 #define SDL_HAS_64BIT_TYPE	1
