diff -Naur wildmidi-wildmidi-0.3.11-orig/src/file_io.c wildmidi-wildmidi-0.3.11/src/file_io.c
--- wildmidi-wildmidi-0.3.11-orig/src/file_io.c	2017-02-03 23:50:47.174642650 +0100
+++ wildmidi-wildmidi-0.3.11/src/file_io.c	2017-02-03 23:51:31.661728117 +0100
@@ -129,6 +129,9 @@
 #elif defined(WILDMIDI_AMIGA)
 	BPTR buffer_fd;
 	long filsize;
+#elif defined(_3DS) || defined(GEKKO)
+	int buffer_fd;
+	struct stat buffer_stat;
 #else /* unix builds */
 	int buffer_fd;
 	struct stat buffer_stat;
