diff -Naur harfbuzz-2.6.4-orig/Makefile.am harfbuzz-2.6.4/Makefile.am
--- harfbuzz-2.6.4-orig/Makefile.am	2020-04-02 16:14:54.289540280 +0200
+++ harfbuzz-2.6.4/Makefile.am	2020-04-02 16:15:01.989540255 +0200
@@ -4,7 +4,7 @@
 
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS = src util test docs
+SUBDIRS = src util docs
 
 EXTRA_DIST = \
 	autogen.sh \
diff -Naur harfbuzz-2.6.4-orig/src/Makefile.am harfbuzz-2.6.4/src/Makefile.am
--- harfbuzz-2.6.4-orig/src/Makefile.am	2020-04-02 16:14:54.279540280 +0200
+++ harfbuzz-2.6.4/src/Makefile.am	2020-04-02 16:15:11.479540224 +0200
@@ -328,15 +328,6 @@
 	|| ($(RM) $(srcdir)/harfbuzz.cc; false)
 BUILT_SOURCES += harfbuzz.cc
 
-noinst_PROGRAMS = \
-	main \
-	test \
-	test-buffer-serialize \
-	test-ot-meta \
-	test-ot-name \
-	test-gpos-size-params \
-	test-gsub-would-substitute \
-	$(NULL)
 bin_PROGRAMS =
 
 main_SOURCES = main.cc
@@ -369,7 +360,6 @@
 
 if HAVE_FREETYPE
 if HAVE_CAIRO_FT
-noinst_PROGRAMS += test-ot-color
 test_ot_color_SOURCES = test-ot-color.cc
 test_ot_color_CPPFLAGS = $(HBCFLAGS) $(FREETYPE_CFLAGS) $(CAIRO_FT_CFLAGS)
 test_ot_color_LDADD = libharfbuzz.la $(HBLIBS) $(FREETYPE_LIBS) $(CAIRO_LIBS) $(CAIRO_FT_LIBS)
