diff -Naur libxmp-lite-4.4.1-orig/src/common.h libxmp-lite-4.4.1/src/common.h
--- libxmp-lite-4.4.1-orig/src/common.h	2020-07-13 20:08:30.532749896 +0200
+++ libxmp-lite-4.4.1/src/common.h	2020-07-13 20:08:41.922845946 +0200
@@ -11,7 +11,7 @@
 
 #if defined(__GNUC__) || defined(__clang__)
 #if !defined(WIN32) && !defined(__ANDROID__) && !defined(__APPLE__) && !defined(__AMIGA__) && !defined(B_BEOS_VERSION) && !defined(__ATHEOS__) && !defined(EMSCRIPTEN) && !defined(__MINT__) 
-#define USE_VERSIONED_SYMBOLS
+//#define USE_VERSIONED_SYMBOLS
 #endif
 #endif
 
