diff -Naur expat-2.2.1-orig/lib/xmlparse.c expat-2.2.1/lib/xmlparse.c
--- expat-2.2.1-orig/lib/xmlparse.c	2017-06-20 07:22:55.000000000 -0600
+++ expat-2.2.1/lib/xmlparse.c	2017-06-20 07:23:24.000000000 -0600
@@ -699,7 +699,9 @@
 # include <errno.h>
 
 # if defined(HAVE_GETRANDOM)
+#undef buffer
 #  include <sys/random.h>    /* getrandom */
+#define buffer (parser->m_buffer) 
 # else
 #  include <unistd.h>        /* syscall */
 #  include <sys/syscall.h>   /* SYS_getrandom */
