diff -Naur fluidsynth-2.4.1-orig/CMakeLists.txt fluidsynth-2.4.1/CMakeLists.txt
--- fluidsynth-2.4.1-orig/CMakeLists.txt	2024-12-01 14:27:38.000000000 +0100
+++ fluidsynth-2.4.1/CMakeLists.txt	2024-12-21 21:34:58.813955060 +0100
@@ -355,14 +355,14 @@
 endif ( WIN32 )
 
 # Check for threads and math
-find_package ( Threads REQUIRED )
+#find_package ( Threads REQUIRED )
 
 find_library ( HAS_LIBM NAMES "m" )
 if ( HAS_LIBM )
   set ( MATH_LIBRARY "m" )
 endif ( HAS_LIBM )
 
-set ( LIBFLUID_LIBS ${MATH_LIBRARY} Threads::Threads )
+set ( LIBFLUID_LIBS ${MATH_LIBRARY} )
 
 # IBM OS/2
 unset ( DART_SUPPORT CACHE )
